app/src/main/java/de/westnordost/streetcomplete/overlays/surface/RoadSurfaceOverlayForm.kt

```diff
@@ -0,0 +1,206 @@
+import androidx.core.view.isVisible
+import androidx.core.widget.doAfterTextChanged
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.osm.edits.update_tags.StringMapChangesBuilder
+import de.westnordost.streetcomplete.data.osm.edits.update_tags.UpdateElementTagsAction
+import de.westnordost.streetcomplete.databinding.FragmentOverlayRoadSurfaceSelectBinding
+import de.westnordost.streetcomplete.osm.surface.SingleSurface
+import de.westnordost.streetcomplete.osm.surface.SingleSurfaceInfo
+import de.westnordost.streetcomplete.osm.surface.SingleSurfaceWithNote
+import de.westnordost.streetcomplete.osm.surface.Surface
+import de.westnordost.streetcomplete.osm.surface.SurfaceAnswer
+import de.westnordost.streetcomplete.osm.surface.SurfaceMissing
+import de.westnordost.streetcomplete.osm.surface.SurfaceMissingWithNote
+import de.westnordost.streetcomplete.osm.surface.applyTo
+import de.westnordost.streetcomplete.osm.surface.asItem
+import de.westnordost.streetcomplete.osm.surface.createMainSurfaceStatus
+import de.westnordost.streetcomplete.osm.surface.toItems
+import de.westnordost.streetcomplete.overlays.AbstractOverlayForm
+import de.westnordost.streetcomplete.quests.surface.COMMON_SPECIFIC_PAVED_SURFACES
+import de.westnordost.streetcomplete.quests.surface.COMMON_SPECIFIC_UNPAVED_SURFACES
+import de.westnordost.streetcomplete.quests.surface.DescribeGenericSurfaceDialog
+import de.westnordost.streetcomplete.quests.surface.GENERIC_ROAD_SURFACES
+import de.westnordost.streetcomplete.quests.surface.GROUND_SURFACES
+import de.westnordost.streetcomplete.quests.surface.shouldBeDescribed
+import de.westnordost.streetcomplete.util.ktx.nonBlankTextOrNull
+import de.westnordost.streetcomplete.view.image_select.DisplayItem
+import de.westnordost.streetcomplete.view.image_select.ImageListPickerDialog
+import de.westnordost.streetcomplete.view.image_select.ItemViewHolder
+
+class RoadSurfaceOverlayForm : AbstractOverlayForm() {
+    override val contentLayoutResId = R.layout.fragment_overlay_road_surface_select
+    private val binding by contentViewBinding(FragmentOverlayRoadSurfaceSelectBinding::bind)
+
+    private val itemsPerRow = 2
+    /** items to display. May not be accessed before onCreate */
+    val items: List<DisplayItem<Surface>> = (COMMON_SPECIFIC_PAVED_SURFACES + COMMON_SPECIFIC_UNPAVED_SURFACES + GROUND_SURFACES + GENERIC_ROAD_SURFACES).toItems()
+    private val cellLayoutId: Int = R.layout.cell_icon_select_with_label_below
+    private var originalSurfaceStatus: SingleSurfaceInfo? = null
+
+    private var selectedStatusForMainSurface: DisplayItem<Surface>? = null
+        set(value) {
+            field = value
+            updateSelectedCell()
+        }
+
+    sealed class SingleSurfaceItemInfo
+    data class SingleSurfaceItem(val surface: DisplayItem<Surface>) : SingleSurfaceItemInfo()
+    data class SingleSurfaceItemWithNote(val surface: DisplayItem<Surface>, val note: String) : SingleSurfaceItemInfo()
```

Comment:
Usually such things should be defined outside the class scope if it does not access any private fields within the containing class. 
It can be made `private` either way.

Reply:
In this case it is used immediately below in one section to ferry data out of GUI data collector (maybe there is a smarter way to achieve this?)